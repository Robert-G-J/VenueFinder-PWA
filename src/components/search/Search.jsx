import React, { Component } from "react";
import "./index.css";
import SearchBarContainer from "../../containers/SearchBarContainer";
import SpotList from "../spotList/SpotList";
import { suggestCompletion } from "../../services/fsqAPI";

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      queryParams: {
        ll: null,
        query: null,
        near: null,
        v: "20180101" // versioning- update as often as possible
      },
      queryResponse: [],
      locationText: "",
      latitude: null,
      longitude: null
    };

    // bindings
    this.onChangeSearchField = this.onChangeSearchField.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.handleQueryChange = this.handleQueryChange.bind(this);
  }

  onChangeSearchField(e) {
    this.setState({ query: e.target.value });
  }

  handleClick(e) {
    e.preventDefault();
    // this.searchForVenues().then(res => {
    //   console.log("FSQ response", res.response.groups[0].items);
    //   this.setState({ queryResponse: res.response.groups[0].items });
    // });
  }

  handleQueryChange(e) {
    this.setState({
      queryParams: { ...this.state.queryParams, query: e.target.value }
    });
  }

  render() {
    const {
      queryParams,
      queryResponse,
      locationText,
      latitude,
      longitude
    } = this.state;

    return (
      <div className="search__body">
        <h1>Search Component</h1>
        {/* <SearchLocation
          locationText={locationText}
          latitude={latitude}
          longitude={longitude}
          onLocationTextChange={this.handleLocationTextChange}
          onCoordinatesChange={this.handleCoordinatesChange}
        /> */}
        <SearchBarContainer />
        <SpotList items={queryResponse} />
      </div>
    );
  }
}
export default Search;
