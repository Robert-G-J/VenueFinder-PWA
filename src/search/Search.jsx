import React from "react";
import SearchBarContainer from "../searchbar/SearchBarContainer";
import SpotListContainer from "../spot-list/SpotListContainer";
import "./index.css";

const Search = () => (
  <div className="search__body">
    <h1>Search Component</h1>
    <SearchBarContainer />
    <SpotListContainer />
  </div>
);
export default Search;
