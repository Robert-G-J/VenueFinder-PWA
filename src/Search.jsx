import React, { Component } from "react";
import "./search.css";
import SearchBar from "./SearchBar";
import SearchLocation from "./SearchLocation";
import SpotTable from "./SpotTable";
import { searchByCoords, searchByText } from "./services/fsqExplore";

class Search extends Component {
  constructor(props) {
    super(props);
    // state declaration
    this.state = {
      queryParams: {
        ll: null,
        query: null,
        near: null,
        v: "20180101" // versioning- update as often as possible
      },
      queryResponse: [],
      locationText: "",
      latitude: null,
      longitude: null
    };

    // bindings
    this.onChangeSearchField = this.onChangeSearchField.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.handleLocationTextChange = this.handleLocationTextChange.bind(this);
    this.handleCoordinatesChange = this.handleCoordinatesChange.bind(this);
    this.handleQueryChange = this.handleQueryChange.bind(this);
    this.usingLocation = this.usingLocation.bind(this);
    this.searchForVenues = this.searchForVenues.bind(this);
  }

  onChangeSearchField(e) {
    this.setState({ query: e.target.value });
  }

  usingLocation() {
    return this.state.queryParams.ll != null;
  }

  searchForVenues() {
    if (this.usingLocation()) {
      return searchByCoords(this.state.queryParams);
    } else {
      return searchByText(this.state.queryParams);
    }
  }

  handleClick(e) {
    e.preventDefault();
    this.searchForVenues().then(res => {
      console.log("FSQ response", res.response.groups[0].items);
      this.setState({ queryResponse: res.response.groups[0].items });
    });
  }

  handleQueryChange(e) {
    this.setState({
      queryParams: { ...this.state.queryParams, query: e.target.value }
    });
  }

  handleCoordinatesChange(latitude, longitude) {
    this.setState({ latitude: latitude, longitude: longitude });
    let lat = parseFloat(latitude);
    let lon = parseFloat(longitude);
    this.setState({
      queryParams: { ...this.state.queryParams, ll: `${lat}, ${lon}` }
    });
  }

  handleLocationTextChange(locationText) {
    this.setState({ locationText: locationText });
    this.setState({
      queryParams: { ...this.state.queryParams, near: locationText }
    });
  }

  render() {
    const {
      queryParams,
      queryResponse,
      locationText,
      latitude,
      longitude
    } = this.state;

    return (
      <div className="search__body">
        <h1>Search Component</h1>
        <SearchLocation
          locationText={locationText}
          latitude={latitude}
          longitude={longitude}
          onLocationTextChange={this.handleLocationTextChange}
          onCoordinatesChange={this.handleCoordinatesChange}
        />
        <SearchBar
          query={queryParams.query}
          handleClick={this.handleClick}
          handleQueryChange={this.handleQueryChange}
        />
        <SpotTable items={queryResponse} />
      </div>
    );
  }
}
export default Search;
