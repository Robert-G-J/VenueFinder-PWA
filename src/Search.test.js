import React from "react";
import Search from "./Search";
import renderer from "react-test-renderer";

describe("The search component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
