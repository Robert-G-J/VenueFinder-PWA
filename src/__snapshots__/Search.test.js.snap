// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The search component renders correctly 1`] = `
<div
  className="search__body"
>
  <div
    className="search-input"
  >
    <h1>
      Search Component
    </h1>
    <div>
      <label
        className="search-input__label"
        htmlFor="search-input"
      />
      <input
        onChange={[Function]}
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="spot-table"
  >
    <ul
      className="spot-table__list"
    />
  </div>
</div>
`;
